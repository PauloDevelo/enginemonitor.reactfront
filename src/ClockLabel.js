import React, { Component } from 'react';
import { FormattedDate, FormattedTime } from 'react-intl';

class ClockLabel extends Component {
    constructor(props) {
      super(props);
		this.state = {
			currentDate: new Date()
		}
    }
    
	componentDidMount() {
		this.intervalID = setInterval(
			() => this.setState((prevState, props) => {return { currentDate: new Date() };}),
			1000
		);
	}
	componentWillUnmount() {
		clearInterval(this.intervalID);
	}
	
    render = () => {
		return (
			<div>
				<FormattedDate value={this.state.currentDate} />
				<span> </span>
				<FormattedTime value={this.state.currentDate} hour='numeric' minute='numeric' second='numeric'/>
			</div>
		)
		
	}
}

export default ClockLabel;
