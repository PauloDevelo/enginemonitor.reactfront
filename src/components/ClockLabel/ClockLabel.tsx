import React, { useState, useEffect } from 'react';
import { FormattedDate, FormattedTime } from 'react-intl';

const ClockLabel = () => {
  const [currentDate, setCurrentDate] = useState(new Date());

  useEffect(() => {
    const intervalID = setInterval(
      () => setCurrentDate(new Date()),
      1000,
    );
    // Specify how to clean up after this effect:
    return function cleanup() {
      clearInterval(intervalID);
    };
  }, []);

  return (
    <span>
      <FormattedDate value={currentDate} />
      <span> </span>
      <FormattedTime value={currentDate} hour="numeric" minute="numeric" second="numeric" />
    </span>
  );
};

export default ClockLabel;
