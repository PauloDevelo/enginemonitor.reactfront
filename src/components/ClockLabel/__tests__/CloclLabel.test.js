import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';
import ClockLabel from '../ClockLabel';

import ignoredMessages from '../../../testHelpers/MockConsole';
// eslint-disable-next-line no-unused-vars
import timeService from '../../../services/TimeService';

import updateWrapper from '../../../testHelpers/EnzymeHelper';

jest.mock('../../../services/TimeService');

describe('ClocLabel', () => {
  beforeEach(() => {
  });

  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('a test was not wrapped in act');
    ignoredMessages.push('Could not find required `intl` object.');
  });

  afterEach(async () => {
    timeService.getUTCDateTime.mockRestore();
  });

  it('should Render the clock as expected', async (done) => {
    // Arrange
    jest.spyOn(timeService, 'getUTCDateTime').mockImplementation(() => new Date(2019, 10, 24, 20, 51));

    // Act
    const clockLabelWrapper = mount(<IntlProvider locale={navigator.language}><ClockLabel /></IntlProvider>);
    await updateWrapper(clockLabelWrapper);

    // Assert
    expect(clockLabelWrapper).toMatchSnapshot();

    expect(timeService.getUTCDateTime).toBeCalledTimes(1);

    clockLabelWrapper.unmount();
    done();
  });
});
