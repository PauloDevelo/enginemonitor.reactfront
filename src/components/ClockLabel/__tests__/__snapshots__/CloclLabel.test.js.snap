// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClocLabel should Render the clock as expected 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <ClockLabel>
    <span>
      <FormattedDate
        value={2019-11-24T12:51:00.000Z}
      >
        11/24/2019
      </FormattedDate>
      <span>
         
      </span>
      <FormattedTime
        hour="numeric"
        minute="numeric"
        second="numeric"
        value={2019-11-24T12:51:00.000Z}
      >
        8:51:00 PM
      </FormattedTime>
    </span>
  </ClockLabel>
</IntlProvider>
`;
